File: Backend/Models/CartItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MealMate.Api.Models
{
    public class CartItem
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public Guid CustomerId { get; set; }
        [ForeignKey("CustomerId")]
        public User? Customer { get; set; }

        [Required]
        public Guid MealId { get; set; }
        [ForeignKey("MealId")]
        public Meal? Meal { get; set; }

        public int Quantity { get; set; }
        
        [Column(TypeName = "decimal(18,2)")]
        public decimal Total { get; set; }
    }
}

File: Backend-Laravel/app/Models/CartItem.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

/**
 * Maps to the "CartItems" table shared with the .NET/EF Core backend
 * (see Backend/Models/CartItem.cs).
 */
class CartItem extends Model
{
    protected $table = 'CartItems';
    protected $primaryKey = 'Id';
    public $incrementing = false;
    protected $keyType = 'string';
    public $timestamps = false;

    protected $fillable = ['Id', 'CustomerId', 'MealId', 'Quantity', 'Total'];

    protected $casts = [
        'Total' => 'decimal:2',
    ];

    protected static function booted(): void
    {
        static::creating(function (CartItem $item) {
            if (empty($item->{$item->getKeyName()})) {
                $item->{$item->getKeyName()} = (string) Str::uuid();
            }
        });
    }

    public function meal()
    {
        return $this->belongsTo(Meal::class, 'MealId', 'Id');
    }

    public function customer()
    {
        return $this->belongsTo(User::class, 'CustomerId', 'Id');
    }
}
